src/script/auth/AuthRepository.ts

```diff
@@ -123,66 +109,45 @@ export class AuthRepository {
             this.logger.warn(`Session expired on access token refresh: ${message}`, error);
-            Raygun.send(error);
+            window.Raygun.send(error);
             return amplify.publish(WebAppEvents.LIFECYCLE.SIGN_OUT, SIGN_OUT_REASON.SESSION_EXPIRED, false);
           }
 
           this.queueState(QUEUE_STATE.READY);
           this.logger.error(`Refreshing access token failed: '${type}'`, error);
-          amplify.publish(WebAppEvents.WARNING.SHOW, WarningsViewModel.TYPE.CONNECTIVITY_RECONNECT);
+          return amplify.publish(WebAppEvents.WARNING.SHOW, WarningsViewModel.TYPE.CONNECTIVITY_RECONNECT);
         });
     }
   }
 
-  /**
-   * Deletes all access token data stored on the client.
-   * @returns {undefined} No return value
-   */
-  deleteAccessToken() {
+  deleteAccessToken(): void {
     resetStoreValue(StorageKey.AUTH.ACCESS_TOKEN.VALUE);
     resetStoreValue(StorageKey.AUTH.ACCESS_TOKEN.EXPIRATION);
     resetStoreValue(StorageKey.AUTH.ACCESS_TOKEN.TTL);
     resetStoreValue(StorageKey.AUTH.ACCESS_TOKEN.TYPE);
   }
 
-  /**
-   * Get the cached access token from the Amplify store.
-   * @returns {Promise} Resolves when the access token was retrieved
-   */
-  getCachedAccessToken() {
+  getCachedAccessToken(): Promise<any> {
     return new Promise((resolve, reject) => {
-      const accessToken = loadValue(StorageKey.AUTH.ACCESS_TOKEN.VALUE);
-      const accessTokenType = loadValue(StorageKey.AUTH.ACCESS_TOKEN.TYPE);
+      const accessToken = loadValue<string>(StorageKey.AUTH.ACCESS_TOKEN.VALUE);
+      const accessTokenType = loadValue<string>(StorageKey.AUTH.ACCESS_TOKEN.TYPE);
 
       if (accessToken) {
         this.logger.info('Cached access token found in Local Storage', {accessToken});
         this.authService.saveAccessTokenInClient(accessTokenType, accessToken);
-        this._scheduleTokenRefresh(loadValue(StorageKey.AUTH.ACCESS_TOKEN.EXPIRATION));
+        this.scheduleTokenRefresh(loadValue(StorageKey.AUTH.ACCESS_TOKEN.EXPIRATION));
         return resolve();
       }
 
       return reject(new z.error.AccessTokenError(z.error.AccessTokenError.TYPE.NOT_FOUND_IN_CACHE));
     });
   }
 
-  /**
-   * Initially get access-token provided a valid cookie.
-   * @returns {Promise} Resolves with the access token data
-   */
-  getAccessToken() {
+  getAccessToken(): Promise<any> {
     return this.authService.postAccess().then(accessToken => this.saveAccessToken(accessToken));
   }
 
-  /**
-   * Store the access token using Amplify.
-   *
-   * @param {Object} accessTokenResponse - Access token data structure
-   * @param {string} accessTokenResponse.access_token - Access token
-   * @param {string} accessTokenResponse.expires_in - Expiration of access token in seconds
-   * @param {string} accessTokenResponse.token_type - Type of access token
-   * @returns {Object} Access token data
-   */
-  saveAccessToken(accessTokenResponse) {
+  saveAccessToken(accessTokenResponse: any): AccessTokenData {
```

Comment:
Let's either keep the JSDoc info about the object or create / use the appropriate interface.